### PR TITLE
rclone: update to fuse3

### LINKED ISSUE #291268: rclone uses fuse3 since v1.62.0 (2023-03-14)
### Describe the bug
rclone wrapper program in [pkgs/applications/networking/sync/rclone/default.nix](https://github.com/NixOS/nixpkgs/blob/7794d8a78570335f10b0f97ae72b14b33c6addc2/pkgs/applications/networking/sync/rclone/default.nix#L53) adds fuse 2.x to PATH, while rclone has seemingly moved over to use fuse 3.x. This seem to break `rclone mount` in certain cases when it's looking for `fusermount3`.

### Steps To Reproduce
I'm trying to emulate whatever `mount -t rclone` does:

```console
$ nix shell github:nixos/nixpkgs/nixpkgs-unstable#rclone
$ tail -F ~/rclone.log
[1] 1024832
tail: cannot open '/home/chris/rclone.log' for reading: No such file or directory
$ env -i RCLONE_DAEMON=true RCLONE_ALLOW_OTHER=true RCLONE_LOG_LEVEL=DEBUG RCLONE_LOG_FILE=/home/chris/rclone.log RCLONE_CONFIG=/etc/rclone.conf RCLONE_CACHE_DIR=/var/rclone /nix/store/f5hqbjla0k1s2dc3kcgbdyw6gjdis65n-rclone-1.65.2/bin/mount.rclone gdrive: /home/chris/Google\ Drive
2024/02/25 00:31:17 DEBUG : Home directory lookup failed and cannot be used as configuration location: exec: "getent": executable file not found in $PATH
2024/02/25 00:31:17 DEBUG : Failed to find user cache dir, using temporary directory: neither $XDG_CACHE_HOME nor $HOME are defined
2024/02/25 00:31:17 DEBUG : Setting --daemon "true" from environment variable RCLONE_DAEMON="true"
2024/02/25 00:31:17 DEBUG : Setting --allow-other "true" from environment variable RCLONE_ALLOW_OTHER="true"
2024/02/25 00:31:17 DEBUG : Setting --daemon "true" from environment variable RCLONE_DAEMON="true"
2024/02/25 00:31:17 DEBUG : Setting --allow-other "true" from environment variable RCLONE_ALLOW_OTHER="true"
2024/02/25 00:31:17 DEBUG : Setting --daemon "true" from environment variable RCLONE_DAEMON="true"
2024/02/25 00:31:17 DEBUG : Setting --allow-other "true" from environment variable RCLONE_ALLOW_OTHER="true"
2024/02/25 00:31:17 DEBUG : Setting --daemon "true" from environment variable RCLONE_DAEMON="true"
2024/02/25 00:31:17 DEBUG : Setting --allow-other "true" from environment variable RCLONE_ALLOW_OTHER="true"
2024/02/25 00:31:17 DEBUG : Setting --daemon "true" from environment variable RCLONE_DAEMON="true"
2024/02/25 00:31:17 DEBUG : Setting --allow-other "true" from environment variable RCLONE_ALLOW_OTHER="true"
2024/02/25 00:31:17 DEBUG : Setting --config "/etc/rclone.conf" from environment variable RCLONE_CONFIG="/etc/rclone.conf"
2024/02/25 00:31:17 DEBUG : Setting --cache-dir "/var/rclone" from environment variable RCLONE_CACHE_DIR="/var/rclone"
2024/02/25 00:31:17 DEBUG : Setting --log-level "DEBUG" from environment variable RCLONE_LOG_LEVEL="DEBUG"
2024/02/25 00:31:17 DEBUG : Setting --log-file "/home/chris/rclone.log" from environment variable RCLONE_LOG_FILE="/home/chris/rclone.log"
tail: '/home/chris/rclone.log' has appeared;  following new file
2024/02/25 00:33:33 DEBUG : rclone: Version "1.65.2" starting with parameters ["/nix/store/f5hqbjla0k1s2dc3kcgbdyw6gjdis65n-rclone-1.65.2/bin/mount.rclone" "mount" "gdrive:" "/home/chris/Google Drive" "--daemon"]
2024/02/25 00:33:33 DEBUG : Creating backend with remote "gdrive:"
2024/02/25 00:33:33 DEBUG : Using config file from "/etc/rclone.conf"
2024/02/25 00:33:34 DEBUG : Google drive root '': 'root_folder_id = 0AFATgfQBaiAeUk9PVA' - save this in the config to speed up startup
2024/02/25 00:33:34 DEBUG : rclone: Version "1.65.2" starting with parameters ["/nix/store/f5hqbjla0k1s2dc3kcgbdyw6gjdis65n-rclone-1.65.2/bin/.rclone-wrapped" "mount" "gdrive:" "/home/chris/Google Drive" "--daemon"]
2024/02/25 00:33:34 DEBUG : Creating backend with remote "gdrive:"
2024/02/25 00:33:34 DEBUG : Using config file from "/etc/rclone.conf"
2024/02/25 00:33:34 DEBUG : Google drive root '': 'root_folder_id = 0AFATgfQBaiAeUk9PVA' - save this in the config to speed up startup
2024/02/25 00:33:34 DEBUG : Google drive root '': Mounting on "/home/chris/Google Drive"
2024/02/25 00:33:34 Fatal error: failed to mount FUSE fs: fusermount: exec: "fusermount3": executable file not found in $PATH
```

An easier example might be to use `systemd.mount` on NixOS:

```nix
{ config, pkgs, lib, ... }:
{
  # Ensure you've got /etc/rclone.conf setup before trying this

  system.activationScripts.makeRcloneDir = lib.stringAfter [ "var" ] ''
    mkdir -p /var/rclone
  '';
  environment.systemPackages = [
    pkgs.rclone
    pkgs.fuse3 # Doesn't help tho
  ];

  systemd.mounts = [
    {
      what = "gdrive:";
      where = "/mnt/gdrive";
      type = "rclone";
      options = "rw,_netdev,allow_other,args2env,vfs-cache-mode=writes,config=/etc/rclone.conf,cache-dir=/var/rclone,log-file=/var/rclone/log.txt,log-level=DEBUG";
    }
  ];
}
```

### Expected behavior
I expect rclone to find `fusermount3` by having it as a dependency.

### Screenshots
N/A

### Additional context
[rclone uses fuse3 since v1.62.0 (2023-03-14):](https://rclone.org/changelog/#v1-62-0-2023-03-14)
> Update to fuse3 after bazil.org/fuse update (Nick Craig-Wood)

### Notify maintainers
@marsam @SuperSandro2000 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.10, NixOS, 23.11 (Tapir), 23.11.20240223.c5101e4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-23.05"`
 - channels(chris): `""`
 - nixpkgs: `/nix/store/80v3x99d9cl7h9fbhqrpajwg4vjyxg6y-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by SuperSandro2000**
This was forgotten in the 1.62.0 update
see https://rclone.org/changelog/#v1-62-0-2023-03-14

Closes #291268

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/sync/rclone/default.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, buildGoModule, fetchFromGitHub, buildPackages, installShellFiles
 , makeWrapper
-, enableCmount ? true, fuse, macfuse-stubs
+, enableCmount ? true, fuse, fuse3, macfuse-stubs
```
Comment (marsam) [inline pkgs/applications/networking/sync/rclone/default.nix:3]: Shall we also change it here?
```suggestion
, enableCmount ? true, fuse3, macfuse-stubs
```

Comment (SuperSandro2000) [reply pkgs/applications/networking/sync/rclone/default.nix:3]: I tried that but then the cgo fuse part is no longer compiling. See the link to the workflow, too.

**Comment by SuperSandro2000** [comment]
At the time they used a mix of both headers through some 3rd party dependency. Maybe that changed in the meantime.

**Comment by SuperSandro2000** [comment]
yep, that looks right

**Pull request merged**